hazelcast-client/src/main/java/com/hazelcast/client/impl/statistics/Statistics.java

```diff
@@ -371,8 +371,19 @@ void fillMetrics(final StringBuilder stats, final ClientConnection ownerConnecti
             }
 
             for (Map.Entry<String, Gauge> entry : allMetrics.entrySet()) {
-                stats.append(STAT_SEPARATOR).append(entry.getKey()).append(KEY_VALUE_SEPARATOR);
-                entry.getValue().render(stats);
+                String key = entry.getKey();
+                stats.append(STAT_SEPARATOR).append(key).append(KEY_VALUE_SEPARATOR);
+
+                Gauge value = entry.getValue();
+                if (value == null) {
+                    value = metricsRegistry.newGauge(key);
```

Comment:
it's still failing for me:
```
java.lang.NullPointerException
    at com.hazelcast.internal.metrics.impl.MetricsRegistryImpl.newGauge(MetricsRegistryImpl.java:238)
    at com.hazelcast.client.impl.statistics.Statistics$PeriodicStatistics.<init>(Statistics.java:353)
    at com.hazelcast.client.impl.statistics.Statistics.start(Statistics.java:113)
    at com.hazelcast.client.impl.HazelcastClientInstanceImpl.start(HazelcastClientInstanceImpl.java:439)
    at com.hazelcast.client.test.TestHazelcastFactory.newHazelcastClient(TestHazelcastFactory.java:76)
    at com.hazelcast.client.statistics.ClientStatisticsTest.createHazelcastClient(ClientStatisticsTest.java:278)
    at com.hazelcast.client.statistics.ClientStatisticsTest.testStatisticsCollectionNonDefaultPeriod(ClientStatisticsTest.java:79)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:95)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
    at java.lang.reflect.Method.invoke(Method.java:508)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:105)
    at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:97)
    at java.util.concurrent.FutureTask.run(FutureTask.java:277)
    at java.lang.Thread.run(Thread.java:785)
```

I think there is no good way to have a generic `newGauge()` method at this moment.
You always have to call `newLongGuage()` or `newDoubleGuage()` Then it's guaranteed you will always get a gauge. Even when the probe does not exist (yet)

Obviously it's failing on IBM JVM only